Source/DiabloUI/diabloui.cpp

```diff
@@ -657,24 +696,17 @@ bool UiValidPlayerName(string_view name)
 	if (!std::all_of(name.begin(), name.end(), IsBasicLatin))
 		return false;
 
-	string_view bannedNames[] = {
-		"gvdl",
-		"dvou",
-		"tiju",
-		"cjudi",
-		"bttipmf",
-		"ojhhfs",
-		"cmj{{bse",
-		"benjo",
-	};
-
 	std::string buffer { name };
 	for (char &character : buffer)
 		character++;
 
-	string_view tempName { buffer };
-	for (string_view bannedName : bannedNames) {
-		if (tempName.find(bannedName) != tempName.npos)
+	AsciiStrToLower(buffer);
+
+	for (string_view bannedName : BannedNames) {
+		std::string loweredBanned = std::string(bannedName);
+		AsciiStrToLower(loweredBanned);

```

Comment:
The `BannedNames` list is already lowercase. You need to call this on `buffer` before the for loop that increments the characters.